numba/smartarray.py

```diff
@@ -92,33 +92,27 @@ def ndim(self): return self._ndim
     @property
     def size(self): return self._size
 
-    def host(self):
-        """Return the host representation of 'self'."""
+    def get(self, where='host'):
+        """Return the representation of 'self' in the given memory space."""
 
-        self._sync('host')
-        return self._host
-
-    def host_changed(self):
-        """Mark the host array as changed, broadcast updates if needed."""
-
-        self._invalidate('gpu')
-        # only sync if there are active views
-        if self._gpu is not None and sys.getrefcount(self._gpu) > 2:
-            self._sync('gpu')
+        self._sync(where)
+        if where == 'host': return self._host
+        elif where == 'gpu': return self._gpu
+        else: raise ValueError('unknown memory space "%s"'%where)
 
-    def gpu(self):
-        """Return the GPU representation of 'self'."""
+    def mark_changed(self, where='host'):
+        """Mark the given location as changed, broadcast updates if needed."""
 
-        self._sync('gpu')
-        return self._gpu
-
-    def gpu_changed(self):
-        """Mark the gpu array as changed, broadcast updates if needed."""
-
-        self._invalidate('host')
-        # only sync if there are active views
-        if self._host is not None and sys.getrefcount(self._host) > 2:
-            self._sync('host')
+        if where == 'host':
+            self._invalidate('gpu')
+            # only sync if there are active views
+            if self._gpu is not None and sys.getrefcount(self._gpu) > 2:
+                self._sync('gpu')
+        elif where == 'gpu':
+            self._invalidate('host')
+            # only sync if there are active views
+            if self._host is not None and sys.getrefcount(self._host) > 2:
+                self._sync('host')
```

Comment:
I shall raise a ValueError throughout that module if 'where' is not a valid target.